db/models.py

```diff
@@ -46,3 +51,61 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class User(AbstractUser):
+    pass
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return f"{self.created_at}"
+
+
+class Ticket(models.Model):
+    movie_session = models.ForeignKey(MovieSession, on_delete=models.CASCADE)
+    order = models.ForeignKey(Order, on_delete=models.CASCADE)
+    row = models.IntegerField()
+    seat = models.IntegerField()
+
+    class Meta:
+        constraints = [
+            UniqueConstraint(
+                fields=["row", "seat", "movie_session"],
+                name="ticket_unique_row_seat_session"
+            )
+        ]
+
+    def __str__(self) -> str:
+        return f"{self.movie_session.movie.title} " \
+               f"{self.movie_session.show_time} " \
+               f"(row: {self.row}, seat: {self.seat})"
+
+    def clean(self) -> None:
+        if not (1 <= self.seat <= self.movie_session.cinema_hall.seats_in_row):
+            raise ValidationError({
+                "seat": [f"seat number must be in available range:"
+                         f" (1, seats_in_row): "
+                         f"(1, {self.movie_session.cinema_hall.seats_in_row})"]
+            })
+        if not (1 <= self.row <= self.movie_session.cinema_hall.rows):
+            raise ValidationError({
+                "row": [f"row number must be in available range:"
+                        f" (1, rows): "
+                        f"(1, {self.movie_session.cinema_hall.rows})"]
+            })
+
+    def save(self, force_insert: bool = False,
+             force_update: bool = False,
+             using: Any = None,
+             update_fields: Any = None
```

Comment:
use more declarative typehint
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contains string elements and can be None has typehint Optional[list[str]]

Reply:
this annotation from build in function of Django ORM, should I use Optional[Any] ?

Reply:
use more declarative typehint for "using", "update_fields" instead of Any
example:
https://docs.djangoproject.com/en/4.2/topics/db/multi-db/#moving-an-object-from-one-database-to-another:~:text=Fred%22)%0A%3E%3E%3E-,p.save(using%3D%22first%22),-%23%20(statement%201)